### PR TITLE
hark-chat-hook: Notifications

### PR DESCRIPTION
**Pull request by liam-fitzgerald**
Drafting, pending frontend changes being finished, just wanted get a pass on the hoon real quick. 

EDIT: Now ready to go!

- No mentions? We could get this by scanning the text of the messages, and chat-cli apparently does this anyway, but not implemented here. 

- Chat can now teleport to arbitrary messages by passing `msg` query param of the message number. This produces a highlighting effect on the message being teleported to. This way, clicking on a message in the inbox takes you exactly  to the message. 
Capture d’écran 2020-11-05 à 4 14 14 PM

Demo frontend: 

Capture d’écran 2020-11-05 à 4 14 01 PM

### CODE REVIEW

**Comment by matildepark** [comment]
S'not bad — 

https://github.com/urbit/urbit/blob/c8068e8fcbc366b6c3eee893e58e38a3d4926698/pkg/arvo/app/chat-cli.hoon#L1102-L1108

**Comment by liam-fitzgerald** [comment]
OTOH, mentions won't work the same way between graph store apps and chat. Could be confusing

**Comment by liam-fitzgerald** [comment]
cc: @urcades for design and @tylershuster for chat changes

File: pkg/interface/src/views/apps/chat/components/ChatWindow.tsx [hunk 1]
```diff
@@ -167,8 +172,10 @@ export default class ChatWindow extends Component<ChatWindowProps, ChatWindowSta
   }
 
   scrollToUnread() {
-    const { mailboxSize, unreadCount } = this.props;
-    this.virtualList?.scrollToData(mailboxSize - unreadCount);
+    const { mailboxSize, unreadCount, scrollTo } = this.props;
+    const target = scrollTo || (mailboxSize - unreadCount);
+    console.log('scrolling', target);
```
Comment (tylershuster) [inline pkg/interface/src/views/apps/chat/components/ChatWindow.tsx:177]: oy

Comment (liam-fitzgerald) [reply pkg/interface/src/views/apps/chat/components/ChatWindow.tsx]: that's it, i'm adding a pre-commit hook to check for `console.log`

Comment (tylershuster) [reply pkg/interface/src/views/apps/chat/components/ChatWindow.tsx]: must-have

**Comment by tylershuster** [comment]
Code looks good, will review on moonnet later

**Comment by liam-fitzgerald** [comment]
Should be backward compatible with the existing moonnet, i'll push a glob so you don't have to run the dev server.

**Comment by liam-fitzgerald** [comment]
come and get it! `|ota ~hastuc-dibtux-hastuc-dibtux %kids` 
You'll need to `|start %hark-chat-hook` as well

**Comment by vvisigoth** [comment]
> No mentions? We could get this by scanning the text of the messages, and chat-cli apparently does this anyway, but not implemented here.

Do it live and die a hero

**Comment by matildepark** [comment]
If L stamps this I'll just merge it in.

> come and get it! `|ota ~hastuc-dibtux-hastuc-dibtux %hark-chat`
> You'll need to `|start %hark-chat-hook` as well. Alternatively, just checkout this branch and `|commit` it into your home desk

Why have multiple QA sources ...? We'll just merge it into the testing branch at hark-redux lol

**Comment by vvisigoth** [comment]
yeah ftr I am still waiting for a response from ~hastuc's moon on the OTA

**Comment by matildepark** [comment]
(I wonder if that custom desk is sharable or if you have to configure that in Clay.)

**Comment by vvisigoth** [comment]
if he didn't run `|public %hark-chat` it might not be configured

**Review by tacryt-socryp — APPROVED** [review]

**Comment by tacryt-socryp** [comment]
I think it should auto-start in %hood (you shouldn't have to start it yourself)

**Comment by liam-fitzgerald** [comment]
mentions now supported!

**Pull request merged**